imap_data_access/file_validation.py

```diff
@@ -400,6 +425,80 @@ def is_valid_repointing(input_repointing: str) -> bool:
 class SPICEFilePath(ImapFilePath):
     """Class for building and validating filepaths for SPICE files."""
 
+    # Covers:
+    # Historical Attitude (type: ah.bc)
+    # Predicted Attitude (type: ap.bc)
+    # Spin Files (type: spin.csv)
+    # Repoint Files (type: repoint.csv)
+    attitude_file_pattern = (
+        r"(?P<{}>imap)_"
+        r"(?P<{}>[\d]{{4}})_"
+        r"(?P<{}>[\d]{{3}})_"
+        r"(?P<{}>[\d]{{4}})_"
+        r"(?P<{}>[\d]{{3}})_"
+        r"(?P<{}>[a-zA-Z0-9\-_]+)\."
+        r"(?P<{}>ah.bc|ap.bc|spin.csv|repoint.csv)"
+    ).format(
+        "mission", "year_start", "doy_start", "year_end", "doy_end", "version", "type"
+    )
```

Comment:
On all of these regexes can we avoid the `.format()` call? It is hard to trace with the parenthesis what gets filled in where.

I think we've been doing `start_date` / `start_time` rather than `date_start`, so swapping the order of the names.
```suggestion
        r"(?P<{mission}>imap)_"
        r"(?P<{start_year}>[\d]{{4}})_"
        r"(?P<{start_doy}>[\d]{{3}})_"
        r"(?P<{end_year}>[\d]{{4}})_"
        r"(?P<{end_doy}>[\d]{{3}})_"
        r"(?P<{version}>[a-zA-Z0-9\-_]+)\."
        r"(?P<{type}>ah.bc|ap.bc|spin.csv|repoint.csv)"
    )
```

Reply:
Yeah I definitely think that looks better, but for some reason our other projects use the format thing so I was just copying those. Maybe so you can quickly see what fields are in the regex?